flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/connection/JdbcConnectionProvider.java

```diff
@@ -20,13 +20,40 @@
 import org.apache.flink.annotation.Internal;
 
 import java.sql.Connection;
+import java.sql.SQLException;
 
 /**
  * JDBC connection provider.
  */
 @Internal
 public interface JdbcConnectionProvider {
-	Connection getConnection() throws Exception;
+	/**
+	 * Get existing connection.
+	 *
+	 * @return existing connection
+	 */
+	Connection getExistingConnection();
 
-	Connection reestablishConnection() throws Exception;
+	/**
+	 * Get existing connection or establish an new one if there is none.
+	 *
+	 * @return existing connection or newly established connection
+	 * @throws SQLException sql exception
+	 * @throws ClassNotFoundException driver class not found
+	 */
+	Connection getConnection() throws SQLException, ClassNotFoundException;
```

Comment:
What do you think about rename this method to `getOrCreateConnection` to make the create/establish behavior more explicitly?

Reply:
I had this tendence too and already had a fixup commit in local, I will push that commit soon.